homeassistant/components/hue/__init__.py

```diff
@@ -98,13 +106,50 @@
     return True
 
 
+async def update_listener(hass: core.HomeAssistant, entry: config_entries.ConfigEntry):
+    """Update entry data with selected hue options and reload."""
+    if entry.options.items() < entry.data.items():
+        # Entry update with no changes, abort.
+        return
+
+    # Instead of individually disable all entities from the undesired platforms,
+    # we remove the unselected device types, clearing the device View.
+    add_remotes = entry.options.get(CONF_INCLUDE_HUE_REMOTES)
+    add_sensors = entry.options.get(CONF_INCLUDE_HUE_SENSORS)
+    if not add_remotes or not add_sensors:
+        devices_to_remove = set()
+        entity_registry = await er.async_get_registry(hass)
+        for entity_data in entity_registry.entities.values():
+            if entity_data.platform != DOMAIN:
+                continue
+
+            if (
+                not add_sensors and entity_data.domain in ("binary_sensor", "sensor")
+            ) or (not add_remotes and entity_data.domain == "remote"):
+                devices_to_remove.add(entity_data.device_id)
+
+        if devices_to_remove:
+            device_registry = await dr.async_get_registry(hass)
+            for device_id in devices_to_remove:
+                device_registry.async_remove_device(device_id)
+
+    # Update entry with new selected options and reload.
+    hass.config_entries.async_update_entry(entry, data={**entry.data, **entry.options})
```

Comment:
I don't see any reason why we would store options in data. They are both already stored on the entry.